src/Argument.h

```diff
@@ -28,19 +28,20 @@ struct Argument {
      * If kind == Scalar, then type fully encodes the expected type
      * of the scalar argument.
      *
-     * If kind == Buffer, then type.bytes() should be used to determine
-     * elem_size of the buffer; additionally, type.code *should* reflect
-     * the expected interpretation of the buffer data (e.g. float vs int),
+     * If kind == InputBuffer|OutputBuffer, then type.bytes() should be used
+     * to determine* elem_size of the buffer; additionally, type.code *should*
+     * reflect the expected interpretation of the buffer data (e.g. float vs int),
      * but there is no runtime enforcement of this at present.
      */
     enum Kind {
         Scalar,
```

Comment:
Do we want to replace Scalar with InputScalar, OutputScalar now? Or can we provide the option of output scalars another way?